pages/api/user-signup.ts

```diff
@@ -0,0 +1,64 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import dbConnect from '@/lib/dbConnect';
+import { validateUserSignUpInput } from '../../lib/validation';
+import { generateHashPassword } from '../../lib/hashPassword';
+import {
+  createNewUser,
+  isUserEmailTaken,
+  deleteUsers,
+} from '../../lib/userApi';
+
+export default async function signUpHandler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  // Connect with MongoDB database
+  try {
+    await dbConnect();
+  } catch (e) {
+    console.log('fail to connect', e);
+    // console.error(e);
+  }
+  // // deleteUsers();
+
+  switch (req.method) {
+    case 'POST':
+      try {
+        const user = JSON.parse(req.body);
+        // console.log('user:', user);
+
+        const validationResult = validateUserSignUpInput(user);
+
+        if (!validationResult.success) {
+          return res
+            .status(400)
+            .json({ success: false, message: validationResult.message });
+        }
+
+        if (await isUserEmailTaken(user.email)) {
+          return res.status(400).json({
+            success: false,
+            message: 'A user already used this email',
+          });
+        }
+
+        const hashedPassword = await generateHashPassword(user.password);
+        const dbRes = await createNewUser(
+          user.first_name,
+          user.last_name,
+          user.email,
+          hashedPassword
+        );
+
+        return res.status(201).json({ success: true, data: dbRes });
```

Comment:
According to the [rfc standards](https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.1), a 200 response would return the object created, a 201 response would return a link or reference (such as an id) to the object created. It seems here the object created is returned. You can decide what data you want to return. Also, it seems that the hashed password is also returned in the data property, which seems unnecessary.

Reply:
My bad, because I to make sure the data I created in the database is fine. I am not sure if I should change this, because the person who is doing the integration may need to take a look at the return data.